File: clientes/FormClientes.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

"""

    Nombre: FormClientes.py
    Autor: Lic. Claudio Invernizzi
    Fecha: 15/01/2026
    E-Mail: cinvernizzi@dsgestion.site
    Licencia: GPL
    Proyecto: Presupuesto
    Comentarios: Clase que recibe como parámetro el contenedor de 
                 el formulario de clientes y agrega los elementos 
                 al mismo 

"""

# importamos las librerías
from clases.fuentes import Fuentes
from clientes.EventosClientes import EventosClientes
from PySide6 import QtGui
from PySide6 import QtCore
from PySide6.QtWidgets import QLabel, QVBoxLayout, QHBoxLayout, QLineEdit, QPushButton

class FormClientes:
    """
    
        :Author: Claudio Invernizzi <cinvernizzi@dsgestion.site>

        Definición de la clase 

    """

    def __init__(self, contenedor):
        """

        :author: Claudio Invernizzi <cinvernizzi@dsgestion.site>        
        :param self: la clase contenedora
        :param contenedor: el layout contenedor del formulario 

        Constructor de la clase, recibe como parámetro el layout
        contenedor del formulario 

        """

        # instanciamos la clase de eventos
        self.Eventos = EventosClientes(self)

        # instanciamos la fuente
        fuente = Fuentes()

        cTitulo = QVBoxLayout()
        cTitulo.setAlignment(QtCore.Qt.AlignTop)
        lForm = QLabel("Datos del Cliente")
        lForm.setFont(fuente.normal)
        lForm.setAlignment(QtCore.Qt.AlignTop)
        cTitulo.addWidget(lForm)

        # definimos la primer fila 
        fila1 = QHBoxLayout()
        fila1.setAlignment(QtCore.Qt.AlignTop)

        # agregamos la clave
        lId = QLabel("Id:")
        lId.setFont(fuente.normal)
        fila1.addWidget(lId)
        self.tId = QLineEdit()
        self.tId.setFont(fuente.normal)
        self.tId.setToolTip("Clave del Registro")
        self.tId.setMaximumWidth(40)
        self.tId.setMinimumHeight(30)
        self.tId.setReadOnly(True)
        fila1.addWidget(self.tId)

        # agregamos el nombre
        lNombre = QLabel("Nombre:")
        lNombre.setFont(fuente.normal)
        fila1.addWidget(lNombre)
        self.tNombre = QLineEdit()
        self.tNombre.setFont(fuente.normal)
        self.tNombre.setToolTip("Nombre completo del cliente")
        self.tNombre.setMinimumWidth(200)
        self.tNombre.setMinimumHeight(30)
        fila1.addWidget(self.tNombre)

        # agregamos la dirección 
        lDireccion = QLabel("Domicilio:")
        lDireccion.setFont(fuente.normal)
        fila1.addWidget(lDireccion)
        self.tDireccion = QLineEdit()
        self.tDireccion.setFont(fuente.normal)
        self.tDireccion.setToolTip("Dirección postal del cliente")
        self.tDireccion.setMinimumWidth(200)
        self.tDireccion.setMinimumHeight(30)
        fila1.addWidget(self.tDireccion)

        # agregamos la primer fila 
        cTitulo.addLayout(fila1)

        # definimos la segunda fila 
        fila2 = QHBoxLayout()
        fila2.setAlignment(QtCore.Qt.AlignTop)

        # agregamos la clave tributaria
        lTributaria = QLabel("Id Tributaria:")
        lTributaria.setFont(fuente.normal)
        fila2.addWidget(lTributaria)
        self.tTributaria = QLineEdit()
        self.tTributaria.setFont(fuente.normal)
        self.tTributaria.setToolTip("Clave tributaria del cliente")
        self.tTributaria.setMaximumWidth(120)
        self.tTributaria.setMinimumHeight(30)
        fila2.addWidget(self.tTributaria)

        # agregamos el teléfono 
        lTelefono = QLabel("Teléfono:")
        lTelefono.setFont(fuente.normal)
        fila2.addWidget(lTelefono)
        self.tTelefono = QLineEdit()
        self.tTelefono.setFont(fuente.normal)
        self.tTelefono.setToolTip("Teléfono del cliente")
        self.tTelefono.setMaximumWidth(150)
        self.tTelefono.setMinimumHeight(30)
        fila2.addWidget(self.tTelefono)

        # agregamos el mail 
        lMail = QLabel("E-Mail:")
        lMail.setFont(fuente.normal)
        fila2.addWidget(lMail)
        self.tMail = QLineEdit()
        self.tMail.setFont(fuente.normal)
        self.tMail.setToolTip("Dirección de correo electrónico")
        self.tMail.setMinimumHeight(30)
        fila2.addWidget(self.tMail)

        # agregamos la fecha de alta
        lAlta = QLabel("Alta:")
        lAlta.setFont(fuente.normal)
        fila2.addWidget(lAlta)
        self.tAlta = QLineEdit()
        self.tAlta.setFont(fuente.normal)
        self.tAlta.setToolTip("Fecha de alta del registro")
        self.tAlta.setMaximumWidth(90)
        self.tAlta.setMinimumHeight(30)
        self.tAlta.setReadOnly(True)
        fila2.addWidget(self.tAlta)

        # agregamos la segunda fila 
        cTitulo.addLayout(fila2)

        # definimos la tercer fila 
        fila3 = QHBoxLayout()
        fila3.setAlignment(QtCore.Qt.AlignTop)

        # agregamos un espaciador
        fila3.addStretch()
        
        # agregamos el botón grabar
        self.btnGrabar = QPushButton("Grabar")
        self.btnGrabar.setFixedHeight(30)
        self.btnGrabar.setFixedWidth(120)
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("recursos/grabar.png"))
        self.btnGrabar.setIcon(icon1)
        self.btnGrabar.setToolTip("Graba el registro en la base")
        fila3.addWidget(self.btnGrabar)

        # agregamos el botón cancelar
        self.btnCancelar = QPushButton("Cancelar")
        self.btnCancelar.setFixedWidth(120)
        self.btnCancelar.setFixedHeight(30)
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("recursos/cancelar.png"))
        self.btnCancelar.setIcon(icon2)
        self.btnCancelar.setToolTip("Reinicia el formulario")
        fila3.addWidget(self.btnCancelar)

        # agregamos la tercer fila 
        cTitulo.addLayout(fila3)

        # agregamos al contenedor
        contenedor.addLayout(cTitulo)

    def nuevoCliente(self):
        """

        :author: Claudio Invernizzi <cinvernizzi@dsgestion.site>

        Método llamado al pulsar el botón nuevo cliente que 
        desencadena la limpieza del formulario 

        """

        # llamamos al método de la clase
        self.Eventos.nuevoCliente()

    def grabaCliente(self):
        """
        
        :author: Claudio Invernizzi <cinvernizzi@dsgestion.site>

        Método llamado al pulsar el botón grabar que verifica 
        los datos del formulario 

        """

        # llamamos al método de la clase
        self.Eventos.verificaCliente()

    def cancelaCliente(self):
        """
        
        :author: Claudio Invernizzi <cinvernizzi@dsgestion.site>

        Método llamado al pulsar el botón cancelar que recarga el 
        registro o limpia el formulario según corresponda

        """

        # llamamos al método
        self.Eventos.cancelaCliente()
    
    def verCliente(self, idcliente):
        """

        :author: Claudio Invernizzi <cinvernizzi@dsgestion.site>

        :param idcliente: clave del registro

        Método que recibe como parámetro la clave de un cliente y llama
        al evento para mostrarlo en pantalla 
        
        """

        # llamamos el evento
        self.Eventos.getDatosCliente(idcliente)

File: src/main/java/site/dsgestion/clientes/FormClientes.java
/**
 * 
 * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
 * @date 23/01/2026
 * @name FormClientes
 * @Projecto: Presupuesto
 * @Copyright DsGestion <dsgestion.site>
 * @Licence: GPL
 * Clase que configura el formulario con los datos de los clientes
 * 
 */

// definimos el paquete
package site.dsgestion.clientes;

// importamos las librerías
import java.awt.*;
import javax.swing.*;
import site.dsgestion.dbApi.Fuentes;


/**
 *
 * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
 * 
 * Definición de la clase
 * 
 */
public class FormClientes {
    
    // definimos los controles
    public JTextField tId;             // clave del registro
    public JTextField tNombre;         // nombre del cliente
    public JTextField tDireccion;      // dirección postal
    public JTextField tIdTributaria;   // identificación tributaria
    public JTextField tTelefono;       // número de teléfono
    public JTextField tMail;           // dirección de correo
    public JTextField tFecha;          // fecha de alta
    protected EventosClientes Eventos; // controlador de eventos
    
    /**
     * 
     * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
     * 
     * @param contenedor el panel contenedor
     * 
     * Constructor de la clase, recibe como parámetro el panel contenedor
     * del formulario 
     * 
     */
    public FormClientes(JPanel contenedor){
        
        // definimos las fuentes
        Fuentes Fuente = new Fuentes();
        
        // instanciamos la clase de eventos
        this.Eventos = new EventosClientes(this);
        
        // definimos el layout
        contenedor.setLayout(new BoxLayout(contenedor, BoxLayout.Y_AXIS));
        
        // agregamos el título
        JPanel pTitulo = new JPanel(new FlowLayout(FlowLayout.LEFT, 5, 5));
        JLabel lTitulo = new JLabel("Datos del Cliente");
        lTitulo.setFont(Fuente.Negrita);
        pTitulo.add(lTitulo);
        contenedor.add(pTitulo);
        
        // definimos la primer fila 
        JPanel fila1 = new JPanel(new FlowLayout(FlowLayout.LEFT, 5, 5));
        
        // agregamos la clave
        JLabel lId = new JLabel("Id:");
        lId.setFont(Fuente.Normal);
        fila1.add(lId);
        this.tId = new JTextField();
        this.tId.setFont(Fuente.Normal);
        this.tId.setToolTipText("Clave del registro");
        this.tId.setPreferredSize(new Dimension(40,30));
        this.tId.setEditable(false);
        fila1.add(this.tId);
        
        // agregamos el nombre 
        JLabel lNombre = new JLabel("Nombre:");
        lNombre.setFont(Fuente.Normal);
        fila1.add(lNombre);
        this.tNombre = new JTextField();
        this.tNombre.setFont(Fuente.Normal);
        this.tNombre.setToolTipText("Nombre compledo del cliente");
        this.tNombre.setPreferredSize(new Dimension(290, 30));
        fila1.add(this.tNombre);
        
        // agregamos la dirección 
        JLabel lDireccion = new JLabel("Dirección:");
        lDireccion.setFont(Fuente.Normal);
        fila1.add(lDireccion);
        this.tDireccion = new JTextField();
        this.tDireccion.setFont(Fuente.Normal);
        this.tDireccion.setToolTipText("Dirección postal del cliente");
        this.tDireccion.setPreferredSize(new Dimension(290,30));
        fila1.add(this.tDireccion);
        
        // agregamos la fila al contenedor
        contenedor.add(fila1);
        
        // definimos la segunda fila 
        JPanel fila2 = new JPanel(new FlowLayout(FlowLayout.LEFT, 5, 5));
        
        // agregamos la id tributaria
        JLabel lIdTributaria = new JLabel("Id Tributaria:");
        lIdTributaria.setFont(Fuente.Normal);
        fila2.add(lIdTributaria);
        this.tIdTributaria = new JTextField();
        this.tIdTributaria.setFont(Fuente.Normal);
        this.tIdTributaria.setToolTipText("Clave tributaria del cliente");
        this.tIdTributaria.setPreferredSize(new Dimension(130,30));
        fila2.add(this.tIdTributaria);
        
        // agregamos el mail 
        JLabel lMail = new JLabel("Mail:");
        lMail.setFont(Fuente.Normal);
        fila2.add(lMail);
        this.tMail = new JTextField();
        this.tMail.setFont(Fuente.Normal);
        this.tMail.setToolTipText("Correo electrónico del cliente");
        this.tMail.setPreferredSize(new Dimension(200,30));
        fila2.add(this.tMail);
        
        // agregamos el teléfono 
        JLabel lTelefono = new JLabel("Teléfono:");
        lTelefono.setFont(Fuente.Normal);
        fila2.add(lTelefono);
        this.tTelefono = new JTextField();
        this.tTelefono.setFont(Fuente.Normal);
        this.tTelefono.setToolTipText("Teléfono del cliente");
        this.tTelefono.setPreferredSize(new Dimension(120,30));
        fila2.add(this.tTelefono);
        
        // la fecha de alta
        JLabel lAlta = new JLabel("Alta:");
        lAlta.setFont(Fuente.Normal);
        fila2.add(lAlta);
        this.tFecha = new JTextField();
        this.tFecha.setToolTipText("Fecha de alta del registro");
        this.tFecha.setPreferredSize(new Dimension(90,30));
        this.tFecha.setEditable(false);
        fila2.add(this.tFecha);
        
        // agregamos el panel al contenedor
        contenedor.add(fila2);
        
        // definimos el panel de los botones
        JPanel fila3 = new JPanel(new FlowLayout(FlowLayout.RIGHT, 5, 5));
        
        // agregamos el botón grabar
        JButton btnGrabar = new JButton("Grabar");
        btnGrabar.setFont(Fuente.Normal);
        btnGrabar.setToolTipText("Graba el registro en la base");
        btnGrabar.setPreferredSize(new Dimension(120, 30));
        Icon icon1 = new ImageIcon(getClass().getResource("/imagenes/grabar.png"));
        btnGrabar.setIcon(icon1);
        btnGrabar.addActionListener(e -> Eventos.verificaCliente());
        fila3.add(btnGrabar);
        
        // agregamos el botón cancelar
        JButton btnCancelar = new JButton("Cancelar");
        btnCancelar.setFont(Fuente.Normal);
        btnCancelar.setToolTipText("Reinicia el formulario");
        btnCancelar.setPreferredSize(new Dimension(120, 30));
        Icon icon2 = new ImageIcon(getClass().getResource("/imagenes/cancelar.png"));
        btnCancelar.setIcon(icon2);
        btnCancelar.addActionListener(e -> Eventos.cancelaCliente());
        fila3.add(btnCancelar);
        
        // agregamos la fila al contenedor
        contenedor.add(fila3);
        
    }
    
    /**
     * 
     * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
     * 
     * Método llamado al pulsar sobre el botón nuevo en la grilla de 
     * clientes, solo invoca el método 
     * 
     */
    public void nuevoCliente(){

        // invocamos el método
        this.Eventos.nuevoCliente();

    }

    /**
     * 
     * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
     * 
     * @param idcliente - entero con la clave del registro
     * 
     * Método llamado al pulsar sobre la grilla de clientes 
     * que recibe como parámetro la clave del registro, 
     * simplemente invoca el evento
     * 
     */
    public void getDatosCliente(int idcliente){

        // invocamos 
        this.Eventos.getDatosCliente(idcliente);

    }
    
}
